src/libopensc/card-openpgp.c

```diff
@@ -2781,14 +2781,21 @@ pgp_calculate_and_store_fingerprint(sc_card_t *card, time_t ctime,
 	/* update the blob containing fingerprints (00C5) */
 	sc_log(card->ctx, "Updating fingerprint blob 00C5.");
 	fpseq_blob = pgp_find_blob(card, 0x00C5);
-	if (fpseq_blob == NULL)
-		LOG_TEST_GOTO_ERR(card->ctx, SC_ERROR_OUT_OF_MEMORY, "Cannot find blob 00C5");
+	if (fpseq_blob == NULL) {
+		r = SC_ERROR_OUT_OF_MEMORY;
+		LOG_TEST_GOTO_ERR(card->ctx, r, "Cannot find blob 00C5");
+	}
+	if (20 * key_info->key_id > fpseq_blob->len) {
```

Comment:
This check covers line `p = newdata + 20 * (key_info->key_id - 1);`, but  `memcpy(p, fingerprint, 20);` is accessing another 20 bytes.

Reply:
Turns out the windows compilers are not happy about this signed/unsigned comparison so this is something I will have to fix too.

Reply:
ugh ... all of these are unsigned on any sane platform. Unless the `20*` would covert the u8 to signed or the u8 would be signed on windows. Any ideas?
```
card-openpgp.c(2788): error C2220: warning treated as error - no 'object' file generated
card-openpgp.c(2788): warning C4018: '>': signed/unsigned mismatch
```

Reply:
not sure, isn't maybe `20` (when not `20U`) considered to by signed?

Reply:
Lets try

Reply:
Seems like it helped. So lets wait for the CI to finish.